src/persist-client/src/async_runtime.rs

```diff
@@ -15,27 +15,35 @@ use mz_ore::task::RuntimeExt;
 use tokio::runtime::{Builder, Runtime};
 use tokio::task::JoinHandle;
 
-/// A runtime for running CPU-heavy asynchronous tasks.
+/// An isolated runtime for asynchronous tasks, particularly work
+/// that may be CPU intensive such as encoding/decoding and shard
+/// maintenance.
+///
+/// Using a separate runtime ensures Persist isolates its expensive
+/// workloads on its own OS threads. This ensures that, even if one
+/// of these tasks fails to yield for arbitrarily long, the OS
```

Comment:
> even if one of these tasks fails to yield for arbitrarily long, the OS

let's make super clear that this is unexpected, CYA, etc etc. probably call out something about spawn_blocking in some way, too. there was a ton of confusion around this during/after the incident and we may as well set ourselves up to have a place we can point to if we get the same confusion again later